web/profiles/custom/os2loop/modules/os2loop_question/src/Helper/Helper.php

```diff
@@ -56,8 +57,19 @@ public function alterForm(array &$form, FormStateInterface $form_state, string $
   private function handleTextFormats(array &$form, FormStateInterface $form_state, string $form_id) {
     $currentFormat = $form['os2loop_question_content']['widget'][0]['#format'] ?? NULL;
     $useRichText = $this->config->get('enable_rich_text') || 'os2loop_question_rich_text' === $currentFormat;
-    $form['os2loop_question_content']['widget'][0]['#better_formats']['settings']['allowed_formats'] =
-      $useRichText ? ['os2loop_question_rich_text' => 'os2loop_question_rich_text'] : ['os2loop_question_plain_text' => 'os2loop_question_plain_text'];
+    if ($useRichText) {
+      $form['os2loop_question_content']['widget'][0]['#better_formats']['settings']['allowed_formats'] =
+        ['os2loop_question_rich_text' => 'os2loop_question_rich_text'];
+      $form["os2loop_question_content"]["widget"][0]["#format"] = 'os2loop_question_rich_text';
+    }
+    else {
+      $form['os2loop_question_content']['widget'][0]['#better_formats']['settings']['allowed_formats'] =
+        ['os2loop_question_plain_text' => 'os2loop_question_plain_text'];
+      $form["os2loop_question_content"]["widget"][0]["#format"] = 'os2loop_question_plain_text';
+    }
+
+    $form['os2loop_question_content']['widget']['#after_build'][] =
+      [$this, 'fieldAfterBuild'];
```

Comment:
Do we need something similar for comments (both comments and answers)?

Reply:
Added.